packages/alpha/src/api/documents/documentsApi.ts

```diff
@@ -0,0 +1,290 @@
+  }
+}
+
+class WordWithBoundingBox extends TextWithBoundingBox {
+  constructor(word: DocumentWordElement) {
+    const left = word.characters[0].left;
+    const right = word.characters[word.characters.length - 1].right;
+    let top = word.characters[0].top;
+    let bottom = word.characters[0].bottom;
+    const page = word.characters[0].page;
+    let text = '';
+
+    for (const character of word.characters) {
+      if (character.top < top) {
+        top = character.top;
+      }
+      if (character.bottom > bottom) {
+        bottom = character.bottom;
+      }
+      text += character.text;
+    }
+
+    super(left, right, top, bottom, page, text);
+  }
+}
+
+class LineWithBoundingBox extends TextWithBoundingBox {
+  words: WordWithBoundingBox[];
+
+  constructor(line: DocumentLineElement) {
+    const words = line.words.map((word) => new WordWithBoundingBox(word));
+
+    const left = words[0].left;
+    const right = words[words.length - 1].right;
+    let top = words[0].top;
+    let bottom = words[0].bottom;
+    const page = words[0].page;
+    let text = '';
+
+    for (const word of words) {
+      if (word.top < top) {
+        top = word.top;
+      }
+      if (word.bottom > bottom) {
+        bottom = word.bottom;
+      }
+      if (text.length > 0) {
+        text += ' ';
+      }
+      text += word.text;
+    }
+
+    super(left, right, top, bottom, page, text);
+    this.words = words;
+  }
+}
+
+class LinesWithBoundingBox extends TextWithBoundingBox {
+  lines: LineWithBoundingBox[];
+
+  constructor(lines: LineWithBoundingBox[]) {
+    let left = lines[0].left;
+    let right = lines[0].right;
+    const top = lines[0].top;
+    const bottom = lines[lines.length - 1].bottom;
+    const page = lines[0].page;
+    let text = '';
+
+    for (const line of lines) {
+      if (line.left < left) {
+        left = line.left;
+      }
+      if (line.right > right) {
+        right = line.right;
+      }
+      if (text.length > 0) {
+        text += '\n';
+      }
+      text += line.text;
+    }
+
+    super(left, right, top, bottom, page, text);
+    this.lines = lines;
+  }
+}
+
+class ParagraphWithBoundingBox extends LinesWithBoundingBox {
+  constructor(element: DocumentParagraphElement) {
+    const lines = element.lines.map((line) => new LineWithBoundingBox(line));
+    super(lines);
+  }
+}
+
+class TableOfContentsWithBoundingBox extends LinesWithBoundingBox {
+  constructor(element: DocumentTableOfContentsElement) {
+    const lines = element.lines.map((line) => new LineWithBoundingBox(line));
+    super(lines);
+  }
+}
+
+class ListWithBoundingBox extends LinesWithBoundingBox {
+  constructor(element: DocumentListElement) {
+    const lines = element.lines.map((line) => new LineWithBoundingBox(line));
+    super(lines);
+  }
+}
+
+class TitleWithBoundingBox extends LinesWithBoundingBox {
+  level: number;
+
+  constructor(element: DocumentTitleElement) {
+    const level = element.level || 3;
```

Comment:
why 3, some levels may be eg. 6? I adjusted the document-science package as some extracted titles had a higher level than 3.

Reply:
At the time I wrote this code, 3 was the highest level possible. I could pick a different number, but 1) I think value will in practice always be set, and 2) maybe 3 is a nice average value to use for a default?

Do you have a different number in mind?

Reply:
No, not sure I have any comments that can help here. im not sure what behavior would be expected if no value is present tbh.
maybe just remove it if it's always set?